src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKeyProviderBuilder.java

```diff
@@ -0,0 +1,125 @@
+package com.amazonaws.encryptionsdk.kms;
+
+import com.amazonaws.ClientConfigurationFactory;
+import com.amazonaws.client.AwsSyncClientParams;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.AwsRegionProvider;
+import com.amazonaws.regions.DefaultAwsRegionProviderChain;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.RegionUtils;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.kms.AWSKMSClient;
+import com.amazonaws.services.kms.AWSKMSClientBuilder;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Used to build a {@link KmsMasterKeyProvider} which will use the defaults found in the relevant provider chains.
+ *
+ * If a region is not defined and no region can be found in the region provider chain, then
+ * {@link Regions#DEFAULT_REGION} is used.
+ * <pre>
+ *     KmsMasterKeyProvider keyProvider = KmsMasterKeyProviderBuilder.defaultProvider();
+ * </pre>
+ *
+ * Example, specifying a region:
+ * <pre>
+ *     KmsMasterKeyProvider keyProvider = KmsMasterKeyProviderBuilder.standard()
+ *       .withRegion("us-east-1")
+ *       .build();
+ * </pre>
+ */
+public class KmsMasterKeyProviderBuilder extends AwsClientBuilder<KmsMasterKeyProviderBuilder, KmsMasterKeyProvider> {
```

Comment:
So, one thing I note is that the Python SDK supports using key IDs from multiple regions in a single KMS MKP. I'd prefer to align the Java behavior more closely with the Python behavior in this regard, and we need to make sure the builder API we're creating here is amenable to this.

Reply:
I believe this is supported with [`withKeyIds`](https://github.com/awslabs/aws-encryption-sdk-java/pull/6/files#diff-854ddaccf0f12318e57edc82ddd8ea44R77).

`withKeyIds` does not set a region, so you could use `withDefaultRegion`, add multiple key IDs, and it should create a KMS MKP for the region you set in `withDefaultRegion`.